test/integration/tcp_proxy_integration_test.cc

```diff
@@ -149,11 +158,19 @@ TEST_P(TcpProxyIntegrationTest, TcpProxyUpstreamDisconnect) {
   tcp_client->close();
 
   EXPECT_EQ("world", tcp_client->data());
+  test_server_.reset();
```

Comment:
why do we need the `test_server_.reset()` now, vs before? Would this still work without that?